tests/conftest.py

```diff
@@ -95,8 +105,36 @@ def _import_and_add_jax(xp_available_backends: dict[str, types.ModuleType]) -> N
     msg = f"unsupported array backend: {GLASS_ARRAY_BACKEND}"
     raise ValueError(msg)
 
-# use this as a decorator for tests involving array API compatible functions
-array_api_compatible = pytest.mark.parametrize("xp", xp_available_backends.values())
+
+@pytest.fixture(
+    params=xp_available_backends.values(), ids=xp_available_backends.values()
+)
+def xp(request: pytest.FixtureRequest) -> types.ModuleType:
+    """
+    Fixture for array backend.
+
+    Access array library functions using `xp.` in tests.
+    """
+    return request.param
+
+
+@pytest.fixture
```

Comment:
I tend to always do this with fixtures
```suggestion
@pytest.fixture(scope="session")
```

Reply:
Whoops, just ran the tests locally and everything works. I'm pretty sure I saw an error yesterday, but maybe it was coming from somewhere else. Thanks for the help!